src/features/configTable/configTable.tsx

```diff
@@ -0,0 +1,222 @@
+  window.alert(JSON.stringify(values));
+};
+
+const validateForm = (values: { configData: Employee[] }) => {
+  const errors: ValidationErrorType = { configData: [] };
+  const nameSurnamePairs: { [key: string]: boolean } = {};
+
+  errors.configData = values.configData.map((employee) => {
+    const employeeErrors: any = {};
+    if (!employee.name) {
+      employeeErrors.name = "Required";
+    }
+    if (!employee.surName) {
+      employeeErrors.surName = "Required";
+    }
+
+    const nameSurnamePair = `${employee.name} ${employee.surName}`;
+    if (nameSurnamePairs[nameSurnamePair] && nameSurnamePair != " ") {
+      employeeErrors.name = "Name and surname pair must be unique";
+      employeeErrors.surName = "Name and surname pair must be unique";
+    } else {
+      nameSurnamePairs[nameSurnamePair] = true;
+    }
+
+    return employeeErrors;
+  });
+
+  return errors;
+};
+
+export default function EditableTableExample(props: {
+  title: string;
+  data: Employee[];
+}) {
+  return (
+    <Form
+      onSubmit={onSubmit}
+      initialValues={{ configData: props.data }}
+      mutators={{
+        ...arrayMutators,
+      }}
+      validate={validateForm}
+      render={({
+        handleSubmit,
+        form: {
+          mutators: { push, pop },
+        },
+        form,
+        hasValidationErrors,
+      }: FormRenderProps<{ configData: Employee[] }>) => {
+        const { pristine, submitting, errors } = form.getState();
+        return (
+          <VStack as="form" onSubmit={handleSubmit} spacing={6}>
+            <Heading>{props.title}</Heading>
+            <HStack spacing={4}>
+              <Button
+                rightIcon={<FaPlus />}
+                onClick={() =>
+                  push("configData", {
+                    name: "",
+                    surName: "",
+                    employed: false,
+                  })
+                }
+                colorScheme="green"
+                aria-label="Add Employee"
+              >
+                New Employee
+              </Button>
+              <Button
+                type="submit"
+                isDisabled={submitting || pristine || hasValidationErrors}
+              >
+                Submit
+              </Button>
+              <Button
+                onClick={() => form.reset()}
+                isDisabled={submitting || pristine}
+              >
+                Reset
+              </Button>
+            </HStack>
+            <Table variant="simple">
+              <Thead>
+                <Tr>
+                  <Th>First Name</Th>
+                  <Th>Last Name</Th>
+                  <Th>Employed</Th>
+                  <Th>Actions</Th>
+                </Tr>
+              </Thead>
+              <Tbody>
+                <FieldArray name="configData">
+                  {({ fields }) =>
+                    fields
+                      .map((employee, index) => (
+                        <Tr key={employee}>
```

Comment:
these 3 lines are quite strange.

you have a 'fields' list, each entry is an 'employee' (so employee is a field?) and employee which looks like an 'object' is used as a Key and below as a string. 

either this doesn't work, or the naming of variables is quite hard to follow.